projects/plugins/jetpack/modules/custom-post-types/portfolios.php

```diff
@@ -255,139 +281,177 @@ function deactivation_post_type_support() {
+					'add_new_item'          => esc_html__( 'Add New Project Type', 'jetpack' ),
+					'new_item_name'         => esc_html__( 'New Project Type Name', 'jetpack' ),
+					'parent_item'           => esc_html__( 'Parent Project Type', 'jetpack' ),
+					'parent_item_colon'     => esc_html__( 'Parent Project Type:', 'jetpack' ),
+					'search_items'          => esc_html__( 'Search Project Types', 'jetpack' ),
+					'items_list_navigation' => esc_html__( 'Project type list navigation', 'jetpack' ),
+					'items_list'            => esc_html__( 'Project type list', 'jetpack' ),
+				),
+				'public'            => true,
+				'show_ui'           => true,
+				'show_in_nav_menus' => true,
+				'show_in_rest'      => true,
+				'show_admin_column' => true,
+				'query_var'         => true,
+				'rewrite'           => array( 'slug' => 'project-type' ),
+			)
+		);
+
+		register_taxonomy(
+			self::CUSTOM_TAXONOMY_TAG,
+			self::CUSTOM_POST_TYPE,
+			array(
+				'hierarchical'      => false,
+				'labels'            => array(
+					'name'                       => esc_html__( 'Project Tags', 'jetpack' ),
+					'singular_name'              => esc_html__( 'Project Tag', 'jetpack' ),
+					'menu_name'                  => esc_html__( 'Project Tags', 'jetpack' ),
+					'all_items'                  => esc_html__( 'All Project Tags', 'jetpack' ),
+					'edit_item'                  => esc_html__( 'Edit Project Tag', 'jetpack' ),
+					'view_item'                  => esc_html__( 'View Project Tag', 'jetpack' ),
+					'update_item'                => esc_html__( 'Update Project Tag', 'jetpack' ),
+					'add_new_item'               => esc_html__( 'Add New Project Tag', 'jetpack' ),
+					'new_item_name'              => esc_html__( 'New Project Tag Name', 'jetpack' ),
+					'search_items'               => esc_html__( 'Search Project Tags', 'jetpack' ),
+					'popular_items'              => esc_html__( 'Popular Project Tags', 'jetpack' ),
+					'separate_items_with_commas' => esc_html__( 'Separate tags with commas', 'jetpack' ),
+					'add_or_remove_items'        => esc_html__( 'Add or remove tags', 'jetpack' ),
+					'choose_from_most_used'      => esc_html__( 'Choose from the most used tags', 'jetpack' ),
+					'not_found'                  => esc_html__( 'No tags found.', 'jetpack' ),
+					'items_list_navigation'      => esc_html__( 'Project tag list navigation', 'jetpack' ),
+					'items_list'                 => esc_html__( 'Project tag list', 'jetpack' ),
+				),
+				'public'            => true,
+				'show_ui'           => true,
+				'show_in_nav_menus' => true,
+				'show_in_rest'      => true,
+				'show_admin_column' => true,
+				'query_var'         => true,
+				'rewrite'           => array( 'slug' => 'project-tag' ),
+			)
+		);
 	}
 
 	/**
 	 * Update messages for the Portfolio admin.
+	 *
+	 * @param array $messages Existing post update messages.
 	 */
-	function updated_messages( $messages ) {
+	public function updated_messages( $messages ) {
```

Comment:
Part of me wonders if we shouldn't get rid of all this method. I don't know that it adds a lot of value, to be honest. I would be happy to have second opinions on this.